src/types/mod.rs

```diff
@@ -262,6 +262,15 @@ pub enum UnsolicitedResponse {
     /// sequence numbers 9, 8, 7, 6, and 5.
     // TODO: the spec doesn't seem to say anything about when these may be received as unsolicited?
     Expunge(u32),
+
+    /// An unsolicited METADATA response (https://tools.ietf.org/html/rfc5464#section-4.4.2)
+    /// that reports a change in a server or mailbox annotation.
+    Metadata {
```

Comment:
This is, sadly, a breaking change. But it seems like we'll have to do one soon anyway. See also #163.

Reply:
I am not sure about what I can do regarding this comment, and I have to say that our project has switched to another library, so I have processed the comments I could, but I can't spend more time on this, so you are free to use this code, but without support from our team. I apologize for any inconvenience.

Reply:
So, the way I approached this with another recent change was to simply drop the unsolicited messages, rather than forward them, with the intent that on the next breaking change, we would add the enum variant and forward them then. The alternative is that we keep things the way they are now, and I only merge this when I make the next breaking release (which I suspect won't be in _too_ long).

I appreciate you updating the PR even though you have moved on from the library! As I don't need this myself either, I will probably then let the PR linger until someone else who needs it comes along and can push it over the finish line.